src/signTx.c

```diff
@@ -199,7 +199,9 @@ static inline void advanceStage()
 		ctx->stage = SIGN_STAGE_WITNESSES;
 		initTxWitnessCtx();
 
-		break;
+		if (ctx->numWitnesses != 0) {
+			break;
+		}
```

Comment:
The comment "// intentional fallthrough" applies here.

Reply:
True.